src/org/labkey/test/util/SampleTypeHelper.java

```diff
@@ -309,4 +310,34 @@ public String getDetailsFieldValue(String label)
         Locator loc = Locator.tag("td").withClass("lk-form-label").withText(label + ":").followingSibling("td");
         return loc.findElement(getDriver()).getText();
     }
+
+    public static Boolean setSampleStatusEnabled(boolean enabled)
+    {
+        return ExperimentalFeaturesHelper.setExperimentalFeature(WebTestHelper.getRemoteApiConnection(false), "experimental-sample-status", enabled);
+    }
+    
+    public void addSampleStates(Map<String, String> states)
+    {
+        waitForText("view data");
+        clickAndWait(Locator.linkContainingText("view data"));
+        DataRegionTable drt = new DataRegionTable("query", this);
+        for (Map.Entry<String, String> statePair : states.entrySet())
+        {
+            if (drt.getRowIndex("Label", statePair.getKey()) < 0)
+            {
+                drt.clickInsertNewRow();
+                addSampleState(statePair.getKey(), statePair.getValue());
+            }
+        }
+    }
+
+    public void addSampleState(String label, @Nullable String stateType)
+    {
+        setFormElement(Locator.name("quf_Label"), label);
+        if (stateType != null)
+        {
+            setFormElement(Locator.name("quf_stateType"), stateType);
+        }
+        clickButton("Submit");
```

Comment:
I cannot remember exact flow... does this cause a page load? If so a clickAndWait might be safer here.

Reply:
`clickButton` already has a default wait built in, which seems to do the trick here.